akka-http/src/main/scala/akka/http/javadsl/server/directives/HeaderDirectives.scala

```diff
@@ -34,7 +35,11 @@ abstract class HeaderDirectives extends FutureDirectives {
    * @group header
    */
   def checkSameOrigin(allowed: HttpOriginRange, inner: jf.Supplier[Route]): Route = RouteAdapter {
-    D.checkSameOrigin(allowed.asScala) { inner.get().delegate }
+    val a = allowed.asScala match {
+      case x: `*`.type ⇒ ???
```

Comment:
`allowed` should be narrowed to `HttpOriginRange.Default` cause `checkSameOrigin(*)` makes no sense.